### ISSUE TITLE
Terminal scroll is messed up when it produces output while hidden

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#138679]
This is an old issue, and it might be known, but I couldn't find an issue. I saw it constantly today while working on smoketest.

- Run smoketest
- Hide the panel
- Reopen the panel
- Use the scroll wheel over terminal
- The scroll jumps all the way to the top of the buffer

**Comment by tpgillam** [comment]
I see something similar with VSCode 1.63.0 on MacOS, and have done for a while. I thought I saw an issue on this a month or so ago, but it was marked as fixed (can't find it again now).

It seems reproducible for me whenever there is a terminal with more than one page of output.

1. With terminal open, generate (somehow) some pages of output.
2. Ensure we are scrolled to the bottom of the terminal. Everything behaving sanely now.
3. Hide the terminal, and display it again.

From this state, I notice the following things:

1. Scrolling 'up' has no effect.
2. Scrolling 'down' jumps the position to the top of the buffer
3. After 2, continuing to scroll down will eventually move the scroll-bar thumb to its lowest position, at which point scrolling stops, BUT the display isn't at the end of the output.
4. After 2, the only way I've found to get the terminal to show the end of the output again is to send input to the terminal (e.g. just press Enter).
